### PR TITLE
Ohm 354 filter by time

### PR DESCRIPTION
**Pull request by napiergit**
Before merging, wait for PR #420, to be merged.

### CODE REVIEW

File: app/scripts/controllers/transactions.js [hunk 1]
```diff
@@ -251,10 +252,17 @@ angular.module('openhimConsoleApp')
       // date filter
       filterDateStart = $scope.settings.filter.startDate;
       filterDateEnd = $scope.settings.filter.endDate;
-      if(filterDateStart && filterDateEnd) {
-        var startDate = moment(filterDateStart).format();
-        var endDate = moment(filterDateEnd).endOf('day').format();
-        filtersObject.filters['request.timestamp'] = JSON.stringify( { '$gte': startDate, '$lte': endDate } );
+  
+      if(filterDateStart || filterDateEnd) {
+        var buildDateFilterObject = function(object, comparator, date) {
```
Comment (BMartinos) [inline app/scripts/controllers/transactions.js]: Maybe move this function of the parent function. It gets created every time the 'returnFilters' function is run.

File: app/scripts/controllers/transactions.js [hunk 2]
```diff
@@ -577,6 +594,13 @@ angular.module('openhimConsoleApp')
 
       // execute refresh if no errors
       if($scope.ngError.hasErrors === false) {
+        
+        // Set dateApplied flag if a date filter is being applied
+        if($scope.settings.filter.startDate || $scope.settings.filter.endDate) {
+          $scope.dateApplied = true;
+        } else {
+          $scope.dateApplied = false;
+        }
```
Comment (BMartinos) [inline app/scripts/controllers/transactions.js]: It doesn't look like this code has any value within this function. This check can be made directly in the filtersApplied function (ln: 695)

**Comment by BMartinos** [comment]
Hey @napiergit Its look great! Just have a few minor comments above. Then one other thing i noted was that if i have both date filters applied, and i manually remove one of them, it doesn't trigger the reload. only when i select dates does it trigger.

File: app/scripts/controllers/transactions.js (hunk 2 above)
Comment (napiergit) [inline app/scripts/controllers/transactions.js]: The date filter is only triggered on blur, but the scope variable is set on date selected, resulting in the (Filters Applied) notification being displayed before the filter is applied.

**Comment by BMartinos** [comment]
Great thank @napiergit Everything looks great. Just have a look at the seconds not applying constantly then we should be good to go.

**Comment by BMartinos** [comment]
Thanks @napiergit this is working great! :+1:

**Pull request merged**